src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -427,17 +459,17 @@ public async Task<uint256> RewindAsync()
             {
                 // Check if rewind data is available in local cache. If it is
                 // we can rewind and there is no need to check underlying storage.
-                if (this.cachedRewindDataList.Count > 0)
+                if (this.cachedRewindDataIndex.Count > 0)
                 {
-                    RewindData lastRewindData = this.cachedRewindDataList.Last();
+                    RewindData lastRewindData = this.cachedRewindDataIndex.Last().Value;
```

Comment:
improve this by taking `var item = cachedRewindDataIndex.Last()` then `item.Value` and `item.Key` small improvment.